lte/cloud/go/services/subscriberdb/servicers/subscriberdb_servicer.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servicers
+
+import (
+	"context"
+	"sort"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"magma/lte/cloud/go/lte"
+	lte_protos "magma/lte/cloud/go/protos"
+	"magma/lte/cloud/go/serdes"
+	lte_models "magma/lte/cloud/go/services/lte/obsidian/models"
+	"magma/lte/cloud/go/services/subscriberdb/obsidian/models"
+	"magma/orc8r/cloud/go/services/configurator"
+	"magma/orc8r/lib/go/protos"
+)
+
+type subscriberdbServicer struct{}
+
+func NewSubscriberdbServicer() lte_protos.SubscriberDBCloudServer {
+	return &subscriberdbServicer{}
+}
+
+// ListSubscribers returns a page of subscribers and a token to be used on
+// subsequent requests. The page token specified in the request is used to
+// determine the first subscriber to include in the page. The page size
+// specified in the request determines the maximum number of entities to
+// return. If no page size is specified, the maximum size configured in the
+// configurator service will be returned.
+func (s *subscriberdbServicer) ListSubscribers(ctx context.Context, req *lte_protos.ListSubscribersRequest) (*lte_protos.ListSubscribersResponse, error) {
+	gateway := protos.GetClientGateway(ctx)
+	if gateway == nil {
+		return nil, status.Errorf(codes.PermissionDenied, "missing gateway identity")
+	}
+	if !gateway.Registered() {
+		return nil, status.Errorf(codes.PermissionDenied, "gateway is not registered")
+	}
+	networkID := gateway.NetworkId
+	gatewayID := gateway.LogicalId
+	lc := configurator.EntityLoadCriteria{
+		PageSize:           req.PageSize,
+		PageToken:          req.PageToken,
+		LoadConfig:         true,
+		LoadAssocsToThis:   true,
+		LoadAssocsFromThis: true,
+	}
+	subEnts, nextToken, err := configurator.LoadAllEntitiesOfType(
+		networkID, lte.SubscriberEntityType, lc, serdes.Entity,
+	)
+	if err != nil {
+		return nil, errors.Wrapf(err, "load subscribers in network of gateway %s", networkID)
+	}
+	lteGateway, err := configurator.LoadEntity(
+		networkID, lte.CellularGatewayEntityType, gatewayID,
+		configurator.EntityLoadCriteria{LoadAssocsFromThis: true},
+		serdes.Entity,
+	)
+	if err != nil {
+		return nil, errors.Wrapf(err, "load cellular gateway for gateway %s", gatewayID)
+	}
+	apnsByName, apnResourcesByAPN, err := loadAPNs(lteGateway)
```

Comment:
loadAPNs seems to be dependent only on networkID, do we need to query for lteGateway here ?

Reply:
We need all of the gateway associations:

`gateway.Associations.Filter(lte.APNResourceEntityType).Keys()`

Reply:
Oh ok. Cool. Missed that earlier.